aten/src/ATen/native/cuda/TensorModeKernel.cu

```diff
@@ -188,15 +188,9 @@ void fused_mode(
     case 16:
     case 8:
     case 4:
-    case 2: {
-      if (ceilPowerOf2 > 2 * C10_WARP_SIZE) {
-        handle_fused_mode<128, scalar_t>(
-            grid, self, ti_values, ti_indices, slice_size, slices);
-      } else {
-        handle_fused_mode<2 * C10_WARP_SIZE, scalar_t>(
-            grid, self, ti_values, ti_indices, slice_size, slices);
-      }
-    }
+    case 2:
+      handle_fused_mode<128, scalar_t>(
+          grid, self, ti_values, ti_indices, slice_size, slices);
```

Comment:
Is it correct to reduce the if-else to just the if logic? Shouldn't we just replace `C10_WARP_SIZE` with `at::cuda::warp_size()`?

Reply:
I thought about this one for a while.  I could have easily used `warp_size` here, the runtime query.  The `handle_fused_mode` function takes a `size` template arg and the `static_assert` ensures the `size` is at least `2 * C10_WARP_SIZE`.  

In the `switch(ceilPowerOf2)` statement, the previously handled case was 256, and the next largest case is 128.  For the if statement `if (128 > 2*C10_WARP_SIZE)` -- we know the warp size is going to be 32 or 64.  So it becomes `if (128 > 64)` for warp size 32, and `if (128 > 128)` for warp size 64.  Not terribly useful IMHO.  When `C10_WARP_SIZE` is 64, it will always call `handle_fused_mode<128>` for cases 128 through 2.  My change here merely makes that clear.  The earlier code wasn't much of an optimization to begin with.

Reply:
Perhaps the if statement was wrong in the first place.  Perhaps it should have been `if (ceilPowerOf2 >= 2 * C10_WARP_SIZE)` --- `>=` instead of just `>`.